list/list_resource.go

```diff
@@ -97,6 +98,9 @@ type ListRequest struct {
 	// IncludeResource indicates whether the provider should populate the
 	// [ListResult.Resource] field.
 	IncludeResource bool
+
+	ResourceSchema         fwschema.Schema
+	ResourceIdentitySchema fwschema.Schema
```

Comment:
Interesting... I'm not really sure what to do about this. We've never just returned the schema in the request like this before, they are typically used to pre-populate a response object and in the case of them potentially not existing (like during upgrade), we just don't populate the response object.

Here, the results stream won't be pre-populated, but provider developers are required to pass along the same schema/identity schema with each result field. Maybe that implementation detail can be placed somewhere in framework?

Reply:
Yes, I consider this good-enough for this PR, while still under construction. When we look at handling non-Framework schema types, I think that adds new design information that will be useful.